dependency.go

```diff
@@ -0,0 +1,171 @@
+package testcontainers
+
+import (
+	"context"
+	"fmt"
+	"github.com/docker/docker/api/types/container"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"golang.org/x/exp/slices"
+)
+
+// ContainerDependency represents a reliance that a container has on another container.
+type ContainerDependency struct {
+	Request ContainerRequest
+	EnvKey  string
+	// CallbackFunc is called after the dependency container is started.
+	CallbackFunc func(Container)
+	// KeepAlive determines whether the dependency should be kept alive after the parent container is terminated.
+	KeepAlive bool
```

Comment:
I've added a `KeepAlive` flag since I thought it'd be helpful to define whether or not the dependency should terminate alongside its parent.

I set the default to `true` in `NewContainerDependency` but I'd be interested to hear other's opinions of the flag itself and if `true` is an appropriate default.